functional/kata-monitor/run.sh

```diff
@@ -0,0 +1,245 @@
+# Copyright (c) 2022 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test file will test kata-monitor for basic functionality (retrieve kata sandboxes)
+# It will assume an environment where:
+# - a CRI container manager (container engine) will be up and running
+# - crictl is installed and configured
+# - the kata-monitor binary is available on the host
+#
+
+CRI_RUNTIME=${CRI_RUNTIME:-"crio"}
+RUNTIME=${RUNTIME:-"containerd-shim-kata-v2"}
+KATA_MONITOR_BIN=${KATA_MONITOR_BIN:-$(type -p kata-monitor)}
+IAM=$(whoami)
+TMPATH=$(mktemp -d -t kata-monitor-test-XXXXXXXXX)
+METRICS_FILE="${TMPATH}/metrics.txt"
+MONITOR_LOG_FILE="${TMPATH}/kata-monitor.log"
+CACHE_UPD_TIMEOUT_SEC=${CACHE_UPD_TIMEOUT_SEC:-10}
+PODID=""
+CONID=""
+
+FALSE=1
+TRUE=0
+
+title() {
+	local STEP="$1"
+	echo "* STEP: $STEP"
+}
+
+echo_ok() {
+	local MSG="$1"
+
+	echo "OK: $MSG"
+	echo ""
+}
+
+error_with_msg() {
+	local MSG=${1}
+
+	echo -e "\nERROR: $MSG"
+	echo -e "\nkata-monitor logs:\n----------------"
+	cat "$MONITOR_LOG_FILE"
+	cleanup
+	echo -e "\nkata-monitor testing: FAILED!"
+	exit 1
+}
+
+cleanup() {
+	crictl stop "$CONID" "$VANILLA_CONID"  > /dev/null 2>&1
+	crictl rm "$CONID" "$VANILLA_CONID"    > /dev/null 2>&1
+	crictl stopp "$PODID" "$VANILLA_PODID" > /dev/null 2>&1
+	crictl rmp "$PODID" "$VANILLA_PODID"   > /dev/null 2>&1
+	kill -9 "$KATA_MONITOR_PID" > /dev/null 2>&1
+	rm -rf "$TMPATH"
+}
+
+create_sandbox_json() {
+	local sbfile="$1"
+
+	POD_NAME="$(date | md5sum | cut -d ' ' -f 1)"
+
+	/bin/cat <<OEM >$sbfile
+{
+	"metadata": {
+		"name": "nginx-$POD_NAME",
+		"namespace": "default",
+		"attempt": 1
+	},
+	"logDirectory": "/tmp",
+	"linux": {
+	}
+}
+OEM
+}
+
+create_container_json() {
+	local cntfile="$1"
+
+	/bin/cat <<OEM >$cntfile
+{
+	"metadata": {
+		"name": "busybox"
+	},
+	"image":{
+		"image": "busybox"
+	},
+	"command": [
+		"top"
+	],
+	"log_path":"busybox.log",
+	"linux": {
+	}
+}
+OEM
+}
+
+start_workload() {
+	local runtime=${1}
+	local args=""
+	local sbfile="${TMPATH}/sandbox.json"
+	local cntfile="${TMPATH}/container.json"
+
+	[ -n "$runtime" ] && args="-r $runtime"
+
+	create_sandbox_json "$sbfile"
+	create_container_json "$cntfile"
+
+	PODID=$(crictl runp $args $sbfile)
+	[ $? -ne 0 ] && error_with_msg "cannot create sandbox"
+	CONID=$(crictl create $PODID $cntfile $sbfile)
+	[ $? -ne 0 ] && error_with_msg "cannot create container"
+	crictl start $CONID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot start container $CONID"
+}
+
+stop_workload() {
+	crictl stop $CONID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot stop container $CONID"
+	crictl rm $CONID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot remove container $CONID"
+	crictl stopp $PODID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot stop pod $PODID"
+	crictl rmp $PODID > /dev/null
+	[ $? -ne 0 ] && error_with_msg "cannot remove pod $PODID"
+}
+
+is_sandbox_there() {
+	local podid=${1}
+	local sbs s
+
+	sbs=$(curl -s 127.0.0.1:8090/sandboxes)
```

Comment:
This URL is used multiple times so I'd consider creating a `readonly` global for it.